### PR TITLE
Improper integral does not evaluate #10445

### PR DESCRIPTION
**Pull request by shubhamtibra**
Modified the code to compute the limit of antiderivative when upper-limit is Infinity. #10445

### CODE REVIEW

File: sympy/integrals/integrals.py [hunk 1]
```diff
@@ -520,8 +521,14 @@ def try_meijerg(function, xab):
                         function = antideriv._eval_interval(x, a, b)
                         function = Poly(function, *gens)
                     elif isinstance(antideriv, Add):
```
Comment (jksuom) [inline sympy/integrals/integrals.py]: I think this alternative should be entirely removed. The whole ``antideriv`` could be passed to ``_eval_interval`` as is done after the next ``else:`` statement.

File: sympy/integrals/integrals.py [hunk 2]
```diff
@@ -520,8 +521,14 @@ def try_meijerg(function, xab):
...
+                            c = symbols('c')
+                            function = Add(*[i._eval_interval(x,a,c) for i in
+                                Add.make_args(antideriv)])
+                            function = limit(function, c, b)
+                        else:
+                             function = Add(*[i._eval_interval(x,a,b) for i in
+                                Add.make_args(antideriv)])
                     else:
                         try:
                             function = antideriv._eval_interval(x, a, b)
```
Comment (shubhamtibra) [inline sympy/integrals/integrals.py]: @jksuom Are you talking about this part? Should we remove the `try` then?

**Comment by shubhamtibra** [comment]
One test is failing [link](https://github.com/sympy/sympy/blob/master/sympy/integrals/tests/test_integrals.py#L1059) . I don't know why.

File: sympy/integrals/integrals.py [hunk 3]
```diff
@@ -519,12 +520,14 @@ def try_meijerg(function, xab):
 
                         function = antideriv._eval_interval(x, a, b)
                         function = Poly(function, *gens)
-                    elif isinstance(antideriv, Add):
-                        function = Add(*[i._eval_interval(x,a,b) for i in
-                            Add.make_args(antideriv)])
                     else:
                         try:
-                            function = antideriv._eval_interval(x, a, b)
+                            if b is S.Infinity:
```
Comment (jksuom) [inline sympy/integrals/integrals.py]: This test and the following lines are not necessary as the limit is taken into account in ``_eval_interval``. They seem to change the result in a way that makes some tests fail.

**Comment by jksuom** [comment]
The failures seem to be caused by the lines containing ``c``. You should remove those and then test locally with bin/test (at least bin/test stats).

**Comment by shubhamtibra** [comment]
What should i use instead of `c`?

**Comment by jksuom** [comment]
The issue is not the symbol name. All those lines can be removed. It is not necessary to consider limits at this point.

**Comment by shubhamtibra** [comment]
I did that but now the integral gives `nan`, are we supposed leave the bug then?

**Comment by jksuom** [comment]
Hmm.. That is odd. I get this with your latest PR.
```
In [1]: integrate(1/(1+exp(x)),(x,0,oo))
Out[1]: log(2)
```
Could there be a difference in our systems.

**Comment by shubhamtibra** [comment]
Now this test is failing `assert integrate(f + exp(z), (z, 2, 3)) == integral_f - exp(2) + exp(3)`.

**Comment by jksuom** [comment]
That is true. But that is just an inconvenience, not a real bug. The assert test can be modified, though I may have to look more closely into the associated issue before doing that.

**Comment by jksuom** [comment]
The lines concerned with Add were added in commit 6fe73a1f0b0c7d49d0e5c197ec8265a0f428c265 in order to solve issue #2708, but its impact on integrals like the one in this issue were not considered. It is probably best to add ``@XFAIL`` above the test until issue #2708 is cleared up.

**Comment by shubhamtibra** [comment]
I am still curious why those lines having `c` were causing that error?

**Comment by jksuom** [comment]
Some integrals cannot be evaluated and raise a NotImplementedError exception. In that case the unevaluated integral containing the local variable ``c`` leaks out via the except statement skipping the limit line and corrupts the result.

**Comment by jksuom** [comment]
Thanks! I'll merge this in a few hours if there are no objections.

**Comment by shubhamtibra** [comment]
Thanks for all the help.

**Pull request merged**